src/ansys/acp/core/_tree_objects/layup_mapping_object.py

```diff
@@ -0,0 +1,313 @@
+    base_element_material_handling :
+        Determines how the base material is handled in the area with reinforcing
+        elements. Can be either ``"remove"`` or ``"retain"``.
+        Used only if ``element_technology`` is ``"reinforcing"``.
+    stress_state :
+        Specifies the stress state of the reinforcing elements.
+        Used only if ``element_technology`` is ``"reinforcing"``.
+    base_material :
+        Define the initial material of the solid elements which will be reinforced
+        by the selected plies.
+        Used only if ``element_technology`` is ``"reinforcing"``.
+    base_element_rosettes :
+        List of rosettes to set the coordinate system of the base elements.
+        This is important if the base material has an orthotropic characteristic.
+        Used only if ``element_technology`` is ``"reinforcing"``.
+    base_element_rosette_selection_method :
+        Defines how the coordinate systems are applied for the base elements.
+        Used only if ``element_technology`` is ``"reinforcing"``.
+
+    """
+
+    __slots__: Iterable[str] = tuple()
+
+    _COLLECTION_LABEL = "layup_mapping_objects"
+    _OBJECT_INFO_TYPE = layup_mapping_object_pb2.ObjectInfo
+    _CREATE_REQUEST_TYPE = layup_mapping_object_pb2.CreateRequest
+    _SUPPORTED_SINCE = "25.1"
+
+    def __init__(
+        self,
+        *,
+        name: str = "LayupMappingObject",
+        active: bool = True,
+        element_technology: ElementTechnology = ElementTechnology.LAYERED_ELEMENT,
+        shell_element_sets: Sequence[ElementSet] = (),
+        use_imported_plies: bool = False,
+        select_all_plies: bool = True,
+        sequences: Sequence[
+            ModelingGroup | ModelingPly | ImportedModelingGroup | ImportedModelingPly
+        ] = (),
+        entire_solid_mesh: bool = True,
+        solid_element_sets: Sequence[SolidElementSet] = (),
+        scale_ply_thicknesses: bool = True,
+        void_material: Material | None = None,
+        minimum_void_material_thickness: float = 1e-6,
+        delete_lost_elements: bool = True,
+        filler_material: Material | None = None,
+        rosettes: Sequence[Rosette] = (),
+        rosette_selection_method: LayupMappingRosetteSelectionMethod = LayupMappingRosetteSelectionMethod.MINIMUM_DISTANCE,  # type: ignore # noqa: E501
+        reinforcing_behavior: ReinforcingBehavior = ReinforcingBehavior.TENSION_AND_COMPRESSION,
+        base_element_material_handling: BaseElementMaterialHandlingType = BaseElementMaterialHandlingType.REMOVE,  # noqa: E501
+        stress_state: StressStateType = StressStateType.PLANE_STRESS_STATE_WITH_TRANSVERSE_SHEAR_AND_BENDING_STIFFNESS,  # noqa: E501
+        base_material: Material | None = None,
+        base_element_rosettes: Sequence[Rosette] = (),
+        base_element_rosette_selection_method: LayupMappingRosetteSelectionMethod = LayupMappingRosetteSelectionMethod.MINIMUM_DISTANCE,  # type: ignore # noqa: E501
```

Comment:
The enum / parameter names here also need updating. My suggestion is to rename `BaseElementMaterialHandlingType` to match the parameter, and leave the rest.